code/modules/mob/living/simple_animal/abnormality/he/singing_machine.dm

```diff
@@ -72,54 +97,60 @@ Finally, an abnormality that DOESN'T have to do any fancy movement shit. It's a
 			bonusRed = minceRed // Should be stronger, for when working it is EXTRA dangerous.
 			manual_emote("makes a horrible grinding noise!") // Oh boy, it's mad.
 			playsound(src, 'sound/abnormalities/singingmachine/chew.ogg', 60, 0, 3)
-	else if(get_attribute_level(user, FORTITUDE_ATTRIBUTE) >= 80 || get_attribute_level(user, TEMPERANCE_ATTRIBUTE) < 60)
-		statChecked = 1 // I see you've failed one of the stat checks, but have also chosen not to feed yourself to the machine.
+		return ..()
+
+	// I see you've failed the stat check, but have also chosen not to feed yourself to the machine.
+	if(get_attribute_level(user, TEMPERANCE_ATTRIBUTE) <= 40)
+		statCheckPenalty = 3
+	else if(get_attribute_level(user, FORTITUDE_ATTRIBUTE) >= 80) // The Fortitude check applies one third of the penalty
+		statCheckPenalty = 1
 	return ..()
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/WorkChance(mob/living/carbon/human/user, chance)
-	chance -= statChecked * 15 // Perish.
+	chance -= statCheckPenalty * 5 // Perish. -5% for the Fortitude check, -15% for the Temperance check
 	return ..()
 
 /mob/living/simple_animal/hostile/abnormality/singing_machine/Worktick(mob/living/carbon/human/user)
 	if(bonusRed) // If you have bonus red damage to apply...
 		user.apply_damage(bonusRed, RED_DAMAGE, null, user.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
-		if(bonusRed < 6 && playStatus == 0)	// Should only happen when the machine isn't dealing damage.
-			for(var/mob/living/carbon/human/H in livinginrange(30, src))
-				if(faction_check_mob(H))
-					continue
-				H.adjustSanityLoss(rand(-1,-2))
+		if(bonusRed < 6 && playStatus == SILENT)	// Should only happen when the machine isn't dealing damage.
```

Comment:
Why does this check against 6 as opposed to the defined `minceRed` and `grindRed` variables?

Reply:
I interpreted this wrong, as i thought the red damage was being reduced somewhere else, but actually dosnt this NEVER activate? since if the machine is dealing bonusRed, its also not silent? Will change.

Reply:
Nevermind, it would apply, actually, the bonus red check is irrelevant though.